core/base/src/main/java/org/eclipse/dataspaceconnector/core/system/CoreServicesExtension.java

```diff
@@ -109,9 +109,9 @@ private void registerParser(ServiceExtensionContext context) {
 
     private void addRetryPolicy(ServiceExtensionContext context) {
 
-        var maxRetries = Integer.parseInt(context.getSetting(MAX_RETRIES, String.valueOf(5)));
-        var minBackoff = Integer.parseInt(context.getSetting(BACKOFF_MIN_MILLIS, String.valueOf(500)));
-        var maxBackoff = Integer.parseInt(context.getSetting(BACKOFF_MAX_MILLIS, String.valueOf(10_000)));
+        var maxRetries = context.getSetting(MAX_RETRIES, 5);
+        var minBackoff = context.getSetting(BACKOFF_MIN_MILLIS, 500);
```

Comment:
They were `int` so I left them `int` but yes, they could be long too.
I was thinking about defining a method like 
```
<T> T getSetting(String setting, T defaultValue)
```
so the type would be implicit (defined by the return type or by the default value) but then there will be the need to implement a "parser" for every type, since I saw only `String` and `int` configuration type this seemed too complicated for what we need

Reply:
I saw other `long` config retrievers, maybe that's an indication that we do need this long parser. 
Will update the PR accordingly